GW2EIEvtcParser/EncounterLogic/Raids/W7/PeerlessQadim.cs

```diff
@@ -191,6 +197,72 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
+                    }
+                    foreach (AbstractCastEvent c in etherStrikes)
+                    {
+                        int coneRadius = 2600;
+                        int coneAngle = 60;
+                        start = (int)c.Time;
+                        end = start + 250;
+                        Point3D facing = replay.Rotations.LastOrDefault(x => x.Time <= start + 300);
+                        replay.Decorations.Add(new PieDecoration(false, 0, coneRadius, facing, coneAngle, (start, end), "rgba(255, 100, 0, 0.30)", new AgentConnector(target)));
+                        replay.Decorations.Add(new PieDecoration(true, 0, coneRadius, facing, coneAngle, (start, end), "rgba(255, 100, 0, 0.1)", new AgentConnector(target)));
+                    }
+                    foreach (AbstractCastEvent c in causticChaos)
+                    {
+                        double acceleration = c.Acceleration;
+                        double ratio = 1.0;
+                        if (acceleration > 0)
+                        {
+                            ratio = acceleration * 0.5 + 1;
+                        }
+                        else
+                        {
+                            ratio = acceleration * 0.6 + 1;
+                        }
+                        int chaosLength = 2600;
+                        int chaosWidth = 100;
+                        start = (int)c.Time;
+                        end = (int)c.EndTime;
+                        int aimTime = (int)((double)c.ExpectedDuration*ratio);
+                        replay.Decorations.Add(new FacingDecoration((0, end), new AgentConnector(target), replay.PolledRotations));
+                        replay.Decorations.Add(new FacingRectangleDecoration((start, end), new AgentConnector(target), replay.PolledRotations, chaosLength, chaosWidth, chaosLength / 2, "rgba(255,100,0,0.3)"));
+                        if (end > start + aimTime)
+                        {
+                            replay.Decorations.Add(new FacingRectangleDecoration((start + aimTime,end), new AgentConnector(target), replay.PolledRotations, chaosLength, chaosWidth, chaosLength / 2, "rgba(100,100,100,0.7)"));
+                        }
+                    }
+                    foreach (AbstractCastEvent c in expoReperc)
+                    {
+                        int radius = 650;
+                        start = (int)c.Time;
+                        end = (int)c.EndTime;
+                        Point3D position = replay.Positions.LastOrDefault(x => x.Time <= start + 1000);
+                        replay.Decorations.Add(new CircleDecoration(true, 0, radius, (start, end), "rgba(255, 220, 0, 0.15)", new PositionConnector(position)));
+                        replay.Decorations.Add(new CircleDecoration(true, end, radius, (start, end), "rgba(255, 220, 50, 0.25)", new PositionConnector(position)));
+
+                        foreach (NPC pylon in TrashMobs.Where(x => x.ID == 21962))
```

Comment:
Wouldn't it make more sense for that to happen while processing pylons in ComputeNPCCombatReplayActors instead of QtP and put the circle decorations inside their combat replay instead of Qadim's?

Reply:
That was my initial idea as well.
The reason why I put it there is that the skill detection has to come through qadim's skill casts anyway since the pylons' skill casts aren't visible. Sadly, this means that the knockback when a pylon is unguarded will also not show up, but I didn't find a way to detect that.

Reply:
Ok, I see. I'll see if there could be a cleaner way to do it in the future, it feels a little bit hacky but I don't see any functional problems to it.

Reply:
Indeed. I didn't like it too much considering its hacky nature but was the best I could do for the moment.